packages/mdc-segmented-button/segment/_segment.scss

```diff
@@ -0,0 +1,157 @@
+// THE SOFTWARE.
+//
+
+// Core: Define complete styles to render component segment, including base and theme styles
+
+@use "@material/button";
+@use "@material/feature-targeting/functions" as feature-targeting-functions;
+@use "@material/feature-targeting/mixins" as feature-targeting-mixins;
+@use './segment-theme';
+
+@mixin core-styles($query: feature-targeting-functions.all()) {
+  $feat-structure: feature-targeting-functions.create-target($query, structure);
+
+  // TODO: include test for element resizing besides the left/right border
+  .mdc-segmented-button__segment {
+    @include segment-theme.base($query);
+    @include unselected($query);
+
+    // Style for when element is selected
+    .mdc-segmented-button__segment--selected {
+      @include selected($query);
+
+      // Avoids double borders
+      + .mdc-segmented-button__segment {
+        @include feature-targeting-mixins.targets($feat-structure) {
+          border-left: hidden;
+        }
+      }
+    }
+
+    &:hover {
+      @include feature-targeting-mixins.targets($feat-structure) {
+        cursor: pointer;
+      }
+    }
+
+    // TODO: verify that this is still needed (cancels ripple focus border)
+    &:focus {
+      @include feature-targeting-mixins.targets($feat-structure) {
+        outline-width: 0;
+      }
+    }
+
+    // TODO: this CSS doesn't work if the buttons are wrapped by divs
+    &:first-child {
+      @include feature-targeting-mixins.targets($feat-structure) {
+        border-radius: segment-theme.$border-radius 0px 0px
+          segment-theme.$border-radius;
+      }
+    }
+
+    &:last-child {
+      @include feature-targeting-mixins.targets($feat-structure) {
+        border-right: visible;
+        border-radius: 0px segment-theme.$border-radius
+          segment-theme.$border-radius 0px;
+      }
+    }
+  }
+
+  .mdc-segmented-button__icon {
+    @include icon($query);
+  }
+
+  .mdc-segmented-button__icon + .mdc-segmented-button__label {
+    @include label($query);
+  }
+
+  .mdc-segmented-button--touch {
+    //TODO
+  }
+
+  .mdc-segmented-button__ripple {
+    //TODO
+  }
+
+  .mdc-segmented-button__touch {
+    //TODO
+  }
+}
+
+@mixin unselected($query: feature-targeting-functions.all()) {
+  $feat-structure: feature-targeting-functions.create-target($query, structure);
+
+  @include button.ink-color(segment-theme.$unselected-ink-color, $query);
+  @include button.outline-color(
+    segment-theme.$unselected-outline-color,
+    $query
+  );
+  @include button.container-fill-color(
+    segment-theme.$unselected-container-fill-color,
+    $query
+  );
+
+  // Avoids double borders
+  @include feature-targeting-mixins.targets($feat-structure) {
+    border-right: hidden;
+  }
+}
+
+@mixin selected($query: feature-targeting-functions.all()) {
+  $feat-structure: feature-targeting-functions.create-target($query, structure);
+
+  @include button.ink-color(segment-theme.$selected-ink-color, $query);
```

Comment:
Since the structure of MDC button and MDC segmented button are different, you should create separate theme mixins for this component.